Shell/NewCNF.cpp

```diff
@@ -1000,53 +999,23 @@ Term* NewCNF::createSkolemTerm(unsigned var, VarSet* free, Formula *reuse_formul
   ASS(domainSorts.isEmpty());
   ASS(fnArgs.isEmpty());
 
-  NameReuse *name_reuse = (env.options->skolemReuse() && reuse_formula)
-    ? NameReuse::skolemInstance()
-    : nullptr;
-  unsigned reused_symbol = 0;
-  bool successfully_reused = false;
-  vstring reuse_key;
-  if(name_reuse) {
-    reuse_key = name_reuse->key(reuse_formula);
-    successfully_reused = name_reuse->get(reuse_key, reused_symbol);
-  }
-  if(successfully_reused)
-    env.statistics->reusedSkolemFunctions++;
-
-  // if we re-use a symbol, we _must_ close over free variables in some fixed order
-  VirtualIterator<unsigned> keyOrderIt;
-  if(name_reuse)
-    keyOrderIt = name_reuse->freeVariablesInKeyOrder(reuse_formula);
-
   VarSet::Iterator vit(*free);
-  while(name_reuse ? keyOrderIt.hasNext() : vit.hasNext()) {
-    unsigned uvar = name_reuse ? keyOrderIt.next() : vit.next();
+  while(vit.hasNext()) {
+    unsigned uvar = vit.next();
     domainSorts.push(_varSorts.get(uvar, AtomicSort::defaultSort()));
     fnArgs.push(TermList(uvar, false));
   }
 
   Term* res;
   bool isPredicate = (rangeSort == AtomicSort::boolSort());
   if (isPredicate) {
-    unsigned pred = reused_symbol;
-    if(!successfully_reused) {
-      pred = Skolem::addSkolemPredicate(arity, domainSorts.begin(), var);
-      if(name_reuse)
-        name_reuse->put(reuse_key, pred);
-      env.statistics->skolemFunctions++;
```

Comment:
I guess the statistics code is factored out somewhere far-far-away?

Reply:
Yes, I moved it back to where it was originally - now at line 1097.